pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImpl.java

```diff
@@ -496,8 +496,13 @@ protected CompletableFuture<Void> doAcknowledge(List<MessageId> messageIdList,
             }
             topicToMessageIdMap.forEach((topicPartitionName, messageIds) -> {
                 ConsumerImpl<T> consumer = consumers.get(topicPartitionName);
-                resultFutures.add(consumer.doAcknowledgeWithTxn(messageIds, ackType, properties, txn)
-                        .thenAccept((res) -> messageIdList.forEach(unAckedMessageTracker::remove)));
+                if (consumer != null) {
+                    resultFutures.add(consumer.doAcknowledgeWithTxn(messageIds, ackType, properties, txn)
+                            .thenAccept((res) -> messageIdList.forEach(unAckedMessageTracker::remove)));
+                } else {
+                    log.warn("MessageIds whose owner topic is {} will be discard because the consumer is not connected",
```

Comment:
Why not be consistent with the behavior of a single ack?
You secretly ignore the message. 
The business thinks that the message has been acked, but it has not, resulting in a ack hole there and Ledger can never be cleared.

Reply:
As I've explained here: https://github.com/apache/pulsar/pull/19874#issuecomment-1477421964

Do you think we should fail the whole ACK if one of them message IDs is invalid?

Reply:
You cannot control how users use your interface. 
We should do interface constraints, not verbal, document, or expect customers can use correctly.
If the interface does not report an exception, it means that it executed successfully, but in fact it is not successful, and the message cannot be cleared. Isn't this a bug?

Reply:
It makes sense to me. I will fix it soon.

Reply:
Fixed. PTAL again @315157973